UnitTests/Views/TextFieldTests.cs

```diff
@@ -2027,6 +2027,27 @@ public void Autocomplete__Added_To_SuperView_On_Add ()
         Assert.Equal (2, superView.Subviews.Count);
     }
 
+    [Fact]
+    public void Right_CursorAtEnd_WithSelection_ShouldClearSelection ()
```

Comment:
Please don't forget to add a unit test for Left:

```cs
    [Fact]
    public void Left_CursorAtStart_WithSelection_ShouldClearSelection ()
```

Reply:
have done